query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -588,22 +605,36 @@ private void handleAggregationAndGroupBy(
   }
 
   private void handleTableFormatResultSet(
-      ResultSetGroup resultSetGroup, List<Builder> rowBuilderList) {
+      ResultSetGroup resultSetGroup,
+      List<Builder> rowBuilderList,
+      LinkedHashSet<String> selectedAttributes,
+      List<String> maskedAttributes) {
     int resultSetGroupCount = resultSetGroup.getResultSetCount();
     for (int i = 0; i < resultSetGroupCount; i++) {
       ResultSet resultSet = resultSetGroup.getResultSet(i);
+      PinotResultAnalyzer resultAnalyzer =
+          PinotResultAnalyzer.create(resultSet, selectedAttributes, viewDefinition);
       for (int rowIdx = 0; rowIdx < resultSet.getRowCount(); rowIdx++) {
         Builder builder;
         builder = Row.newBuilder();
         rowBuilderList.add(builder);
 
-        for (int colIdx = 0; colIdx < resultSet.getColumnCount(); colIdx++) {
+        for (int colIdx = 0, logicalColIdx = 0;
+            colIdx < resultSet.getColumnCount();
+            colIdx++, logicalColIdx++) {
```

Comment:
Why do we need `logicalColIdx`?

Reply:
There can be multiple map fields. When creating the idxToLogical name map, a map field would only increment the counter by one. Here each map field increments colIdx by 2. That's why I have created a new variable, which is incremented only once even if we go through a maps 2 columns (key and value)